docs/source/usage/torch_compile.rst

```diff
@@ -0,0 +1,187 @@
+
+.. code-block:: bash
+
+    $ # running a 8B model on H100 with batch size 1, 36.39 seconds of compilation time, 7.7% improvement in latency
+
+    $ python3 benchmarks/benchmark_latency.py --model meta-llama/Meta-Llama-3-8B --batch-size 1 --load-format dummy
+    init engine (profile, create kv cache, warmup model) took 11.79 seconds
+    Avg latency: 0.9704469823899369 seconds
+
+    $ python3 benchmarks/benchmark_latency.py --model meta-llama/Meta-Llama-3-8B --batch-size 1 --load-format dummy -O "{'level': 3, 'candidate_compile_sizes': [1]}"
+    init engine (profile, create kv cache, warmup model) took 48.18 seconds
+    Avg latency: 0.8950413154981409 seconds
+
+    $ # running a 8B model on L4 with batch size 1, 66.54 seconds of compilation time, 4.1 % improvement in latency
+
+    $ python3 benchmarks/benchmark_latency.py --model meta-llama/Meta-Llama-3-8B --batch-size 1 --load-format dummy
+    init engine (profile, create kv cache, warmup model) took 20.63 seconds
+    Avg latency: 7.81603614680001 seconds
+
+    $ python3 benchmarks/benchmark_latency.py --model meta-llama/Meta-Llama-3-8B --batch-size 1 --load-format dummy -O "{'level': 3, 'candidate_compile_sizes': [1]}"
+    init engine (profile, create kv cache, warmup model) took 87.17 seconds
+    Avg latency: 7.495755991366673 seconds
+
+For a dynamic workload, we can use the ``VLLM_LOG_BATCHSIZE_INTERVAL`` environment variable to monitor the batch size distribution:
+
+.. code-block:: bash
+
+    $ # running an 8B model on H100 with various batch sizes, 72.76 seconds of compilation time, 3.9% improvement in throughput
+    $
+    $ # 1. Run the baseline setting
+    $ python3 benchmarks/benchmark_throughput.py --input-len 256 --output-len 256 --model meta-llama/Meta-Llama-3-8B --load-format dummy --num-scheduler-steps 64
+    init engine (profile, create kv cache, warmup model) took 14.42 seconds
+    Throughput: 44.39 requests/s, 22728.17 total tokens/s, 11364.08 output tokens/s
+
+    $ # 2. Run the same setting with profiling
+    $ VLLM_LOG_BATCHSIZE_INTERVAL=1.0 python3 benchmarks/benchmark_throughput.py --input-len 256 --output-len 256 --model meta-llama/Meta-Llama-3-8B --num-scheduler-steps 64
+    INFO 12-10 15:42:47 forward_context.py:58] Batchsize distribution (batchsize, count): [(256, 769), (232, 215), ...]
+
+    $ # 3. The most common batch sizes are 256 and 232, so we can compile the model for these two batch sizes
+    $ python3 benchmarks/benchmark_throughput.py --input-len 256 --output-len 256 --model meta-llama/Meta-Llama-3-8B --num-scheduler-steps 64 -O "{'level': 3, 'candidate_compile_sizes': [232, 256]}"
+    init engine (profile, create kv cache, warmup model) took 87.18 seconds
+    Throughput: 46.11 requests/s, 23606.51 total tokens/s, 11803.26 output tokens/s
```

Comment:
repeat for llama 3.2 1B on H100, almost no improvement.

```bash
    $ # running an 1B model on H100 with various batch sizes, 39.79 seconds of compilation time, 0.5% improvement in throughput
    $
    $ # 1. Run the baseline setting
    $ python3 benchmarks/benchmark_throughput.py --input-len 256 --output-len 256 --model meta-llama/Llama-3.2-1B --load-format dummy --num-scheduler-steps 64
    init engine (profile, create kv cache, warmup model) took 13.14 seconds
    Throughput: 116.83 requests/s, 59814.48 total tokens/s, 29907.24 output tokens/s

    $ # 2. Run the same setting with profiling
    $ VLLM_LOG_BATCHSIZE_INTERVAL=1.0 python3 benchmarks/benchmark_throughput.py --input-len 256 --output-len 256 --model meta-llama/Llama-3.2-1B --num-scheduler-steps 64
    INFO 12-10 15:42:47 forward_context.py:58] Batchsize distribution (batchsize, count): [(256, 769), (232, 215), ...]

    $ # 3. The most common batch sizes are 256 and 232, so we can compile the model for these two batch sizes
    $ python3 benchmarks/benchmark_throughput.py --input-len 256 --output-len 256 --model meta-llama/Llama-3.2-1B --num-scheduler-steps 64 -O "{'level': 3, 'candidate_compile_sizes': [232, 256]}"
    init engine (profile, create kv cache, warmup model) took 52.93 seconds
    Throughput: 117.38 requests/s, 60100.50 total tokens/s, 30050.25 output tokens/s
```

repeat for llama 3.2 1B on L4 (it's even slower):

```bash
    $ # running an 1B model on L4 with various batch sizes, 58.5 seconds of compilation time, -6.0% improvement in throughput
    $
    $ # 1. Run the baseline setting
    $ python3 benchmarks/benchmark_throughput.py --input-len 256 --output-len 256 --model meta-llama/Llama-3.2-1B --load-format dummy --num-scheduler-steps 64
    init engine (profile, create kv cache, warmup model) took 21.77 seconds
    Throughput: 16.36 requests/s, 8376.21 total tokens/s, 4188.10 output tokens/s

    $ # 2. Run the same setting with profiling
    $ VLLM_LOG_BATCHSIZE_INTERVAL=1.0 python3 benchmarks/benchmark_throughput.py --input-len 256 --output-len 256 --model meta-llama/Llama-3.2-1B --load-format dummy --num-scheduler-steps 64
    INFO 12-10 15:42:47 forward_context.py:58] Batchsize distribution (batchsize, count): [(256, 769), (232, 215), ...]

    $ # 3. The most common batch sizes are 256 and 232, so we can compile the model for these two batch sizes
    $ python3 benchmarks/benchmark_throughput.py --input-len 256 --output-len 256 --model meta-llama/Llama-3.2-1B --load-format dummy --num-scheduler-steps 64 -O "{'level': 3, 'candidate_compile_sizes': [232, 256]}"
    init engine (profile, create kv cache, warmup model) took 80.27 seconds
    Throughput: 15.38 requests/s, 7873.07 total tokens/s, 3936.54 output tokens/s
```